### DISCUSSION TITLE
Looks like VS Codespaces is going under in favor of Github Codespaces

### DISCUSSION THREAD

**Discussion by MarcusOy** [coder/code-server#2077] [General] [answered]
I thought this was quite relevant. As explained in [this blog article](https://devblogs.microsoft.com/visualstudio/visual-studio-codespaces-is-consolidating-into-github-codespaces/#comment-23677), VS Codespaces is being replaced by Github Codespaces. I've been using Codespaces extensively while away from my dev machine using my Macbook, and it works great! It feels really great to use, especially for free. However, it looks like during the planning of the transition, they've decided to cut self-hosting from the feature set of Github Codespaces, as seen [here](https://docs.microsoft.com/en-us/visualstudio/codespaces/resources/consolidation-faq#are-any-features-not-moving-over-to-github). I'm sure a lot of using that were using the feature will be jumping ship to this project. What do you guys think?

**Comment by nhooyr** [comment, marked as answer]
Yup, we're aware and actively working on a feature to help people migrate.

Stay tuned, it should be out within the next 2 weeks!

**Comment by Firehawke** [comment]
Let's hope that will be ready soon, because there are a lot of people who got caught flat-footed by this particular Microsoft gaffe and are pretty angry over having the rug pulled out from under them.

**Comment by bpmct** [comment]
@MarcusOy @Firehawke We are close and actively looking for feedback on this! https://github.com/cdr/code-server/discussions/2137
